d3-selection/index.d.ts

```diff
@@ -12,7 +14,7 @@
  * without 'd3-selection' trying to use properties internally which would otherwise not
  * be supported.
  */
-export type BaseType = Element | EnterElement | Window;
+export type BaseType = Element | EnterElement | Document | Window | null;
```

Comment:
`null` needs to remain part of the `BaseType` for a couple of reasons, most prominently, there are _top-level_ selection methods, which result in the parent element being set to `null`:

E.g. : `select<GElement extends BaseType, OldDatum>(node: GElement): Selection<GElement, OldDatum, null, undefined>;`

Also, it has now become idiomatic to to create a _subselection_ which is empty with `null` elements before performing a first data join (see discussion in [d3-selection Issue 79](https://github.com/d3/d3-selection/issues/79) ). So to stay consistent with this idiomatic use, we have the sub-selection signatures:
```ts
    /**
     * Create an empty sub-selection. Selection.selectAll does affect grouping: The elements in the returned
     * selection are grouped by their corresponding parent node in this selection, the group at the current index will be empty.
     */
    selectAll(): Selection<null, undefined, GElement, Datum>;
    /**
     * Create an empty sub-selection. Selection.selectAll does affect grouping: The elements in the returned
     * selection are grouped by their corresponding parent node in this selection, the group at the current index will be empty.
     */
    selectAll(selector: null): Selection<null, undefined, GElement, Datum>;
    /**
     * Create an empty sub-selection. Selection.selectAll does affect grouping: The elements in the returned
     * selection are grouped by their corresponding parent node in this selection, the group at the current index will be empty.
     */
    selectAll(selector: undefined): Selection<null, undefined, GElement, Datum>;

```

So for this reason, `null` should remain in the `BaseType`.

Note however, that e.g. `d3-drag` and `d3-zoom` have decoupled much stricter base type, I.e. they are currently limited to just `Element`.